### DISCUSSION TITLE
Interpolating between GeoJSON polygons (isochrones)

### DISCUSSION THREAD

**Discussion by pete-murphy** [d3/d3#3938] [Q&A] [answered]
Just trying to get a sense of what the right approach is to this problem: I've got a point _P_ that is between two isochrones (polygons _I_ and _J_, representing say 50-minute walking time and 60-minute walking time from some origin _O_, respectively). I'd like to be able to estimate a more exact walking time to _P_ based on its relative distance between the isochrones.

![untitled (14)](https://github.com/user-attachments/assets/03efb124-4070-4475-92b3-095ea3f71e1a)

One thing that I could do is iterate through all the vertices in _I_ to find the two closest to _P_, and similarly for _J_. Now I've got distance from closest edge in _I_ (call it _D<sub>I</sub>_ :shrug:) and distance from closest edge in _J_ (_D<sub>J</sub>_), and could do: _D<sub>I</sub>_ &div; (_D<sub>I</sub>_ + _D<sub>J</sub>_). Does that sound right? Is there anything in `d3` that would help with this calculation?

Alternatively maybe I could subdivide the isochrone into more granular isochrones since I know my origin points are fixed, it might be faster to cache these sub-isochrones rather than doing this calculation on the fly. I thought maybe something like `geoCentroid` would help with that, but it doesn't seem to...

Made a WIP notebook here https://observablehq.com/d/731b8681e60a9432

**Comment by Fil** [comment, marked as answer]
Can you share your notebook? You're not saying how you're computing the isochrones, but if it's not through an API, I think you'll be better off computing the exact distance to your point, and if it's an API that is not too costly, by requesting more granular results. If you really want to interpolate polygons, you could try a distance transform method (see https://observablehq.com/@climatelab/distance-to-the-border-of-protected-areas and https://observablehq.com/@mourner/fast-distance-transform).

**Reply by pete-murphy** [reply]
Sorry I didn't see a notification about your reply! I've updated the Notebook to be public https://observablehq.com/d/731b8681e60a9432. The isochrones are requested from an API, I think I probably will just request the most granular results I can.
Thanks for the other links, I'll take a look!
